src/compas_timber/ghpython/components/CT_Bake_BoxMap/code.py

```diff
@@ -14,44 +16,47 @@
 
 
 class BakeBoxMap(component):
-    def RunScript(self, model, map_size, bake):
-        if map_size and len(map_size) != 3:
-            self.AddRuntimeMessage(
-                Error, "Input parameter MapSize requires exactly three float values (scale factors in x,y,z directions)"
-            )
-            return
-
-        if map_size:
-            dimx, dimy, dimz = map_size
-        else:
-            # for the pine 251 material bitmap, rotated
-            dimx = 0.2
-            dimy = 0.2
-            dimz = 1.0
+    def RunScript(self, model, beam_map_size, plate_map_size, beam_layer_name, plate_layer_name, swap_uv, bake):
+        def get_dimensions(map_size, default):
+            return map_size if map_size else default
 
+        # Check if model exists and baking is enabled
         if not model:
             self.AddRuntimeMessage(Warning, "Input parameters Model failed to collect any Beam objects.")
             return
 
-        if not bake:
+        if not bake:  # Exit early if bake is False
             return
 
+        # Save current Grasshopper document context
+        ghdoc = sc.doc
```

Comment:
not sure if you need to do that. If I'm not mistaken `ghdoc` is always available in component scope.